### PR TITLE
Create a quantized in-palce version CUDA ReLU function, relu_quantized_cuda_.

### PR DESCRIPTION
**Pull request by fufeisi**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #85670
Summary:
this and #85669 are to allow the relu function to run on a quantized tensor on cuda. That is torch.relu(qa) for a quantized tensor qa on cuda.
Test Plan:
python test/test_quantization.py

Previous PR that has been reverted: #85502.

### CODE REVIEW

**Comment by dzdang** [comment]
can you link the previous PR (the one with the review history) in the summary

**Review by dzdang — APPROVED** [review]

**Review by z-a-f — APPROVED** [review]

File: test/quantization/core/test_quantized_op.py [hunk 1]
```diff
@@ -255,18 +255,6 @@ def test_qrelu(self):
         ]
         devices = ["cpu", "cuda"] if TEST_CUDA else ["cpu"]
         for device in devices:
-            # Only test the non-in-place version relu quantized cuda,
```
Comment (dzdang) [inline test/quantization/core/test_quantized_op.py:258]: thought I left a review comment here but can't find it. I'm wondering why we add this in the previous PR and remove it here?

Comment (fufeisi) [reply test/quantization/core/test_quantized_op.py:258]: For the previous PR, we only have non-in-palce qrelu, so I add this for only test non-in-palce qrelu, avoid in-palce qrelu.

**Comment by fufeisi** [comment]
@mergebot rebase

**Pull request closed without merging**